app/assets/stylesheets/views/_homepage.scss

```diff
@@ -216,108 +216,127 @@ body.homepage {
+        .search-input-wrapper {
+          @include box-sizing(border-box);
 
-        @include ie-lte(7){
-          padding-left: 0;
-        }
-        .js-enabled & {
-          width: 86%;
-          @include calc(width, "100% - 37px");
-        }
-      }
+          padding-right: $search-button-size;
+          margin-right: -$search-button-size;
 
-      input.submit {
-        position: absolute;
-        z-index: 4;
-        right: 0;
-        top: 0;
-        width: 40px;
-        height: 40px;
+          .search-input {
+            @include box-sizing(border-box);
+            @include appearance(none);
 
-        border: 1px solid $black;
-        border-width: 0 0 0 1px;
-        @include ie-lte(7){
-          border-width: 1px;
+            position: relative;
+            height: $search-button-size;
+            margin: 0;
+
+            border: 0;
+            outline-offset: 0;
+          }
         }
-        border-left-color: #222;
-        @include border-radius(0);
 
-        overflow: hidden;
-        text-indent: -5000px;
-        cursor: pointer;
+        .submit {
+          @include box-sizing(border-box);
+          height: $search-button-size;
+          width: $search-button-size;
+          
+          cursor: pointer;
 
-        color: $white;
-        background: $black image-url("homepage/search-button.png") no-repeat 2px 50%;
+          overflow: hidden;
+          text-indent: -5000px;
+          background: $black image-url("homepage/search-button.png") no-repeat 2px 50%;
+          
+          @include device-pixel-ratio() {
+            background-size: 52.5px 35px;
+            background-position: 115% 50%;
+          }
 
-        @include device-pixel-ratio() {
-          background-size: 52.5px 35px;
-          background-position: 115% 50%;
-        }
+          &:hover {
+            background-color: lighten($black, 5%);
+          }
+
+          &:focus {
+            z-index: 15;
+            position: relative;
+          }
 
-        @include ie-lte(7) {
-          background-image: none;
-          width: auto;
+          @include border-radius(0);
         }
 
-        &:hover {
-          background-color: lighten($black, 5%);
+        // If JS is enabled, and browser is not IE6 or IE7, position label
+        // as a placeholder
+        .js-enabled & {
+          position: relative;
+          background: $white;
+
+          // Absolutely position the label behind the input
+          label {
+            @include core-19($line-height: 40px, $line-height-640: 40px);
+            position: absolute;
+            left: 0;
+            top: 1px;
+            z-index: 5;
+
+            text-indent: 15px;
+
+            color: $secondary-text-colour;
+          }
+
+          // Input is transparent and sits above the label
+          .search-input {
+            position: relative;
+            z-index: 10;
+
+            // Internet Explorer does not play well with "empty" elements. The
```

Comment:
Minor: Having the links is much better now and helps to understand the issue. But I think the way it is explained it not great (already on the SO post which you half-copied here). Instead of '"empty" elements' I think it would be clearer to say "form fields without an empty content/value". It is not clear from the original post if it really applies to **all** elements, I would suspect it's only true for form fields. It would also be good to know which IEs are affected and only apply to those. But will probably be difficult to find out. And the code doesn't cause any harm even if it gets applied to versions which have no issues here.
I also like the solution with the data url better than the `#` before.